### DISCUSSION TITLE
Support Unix socket ?

### DISCUSSION THREAD

**Discussion by kinchan** [dani-garcia/vaultwarden#2757] [Q&A] [answered]
Hi, first of all great thank you for project member.

I tried with mysql ,and set below

DATABASE_URL=mysql://vw_user:vw_userpassword@localhost/vaultwarden

It's work good with TCP port(default 3306).

Next I tried Unix socket, about below.

DATABASE_URL=mysql://vw_user:vw_userpassword@localhost:/var/run/mysql/mysql.sock/vaultwarden

But not work.

My question, Vaultwarden support Unix socket ?
Please teach me.

Thank you for reading.

**Comment by BlackDex** [comment, marked as answer]
In our current version there isn't support for sockets.
When we upgrade Diesel to v2.x it should be there.
Then  you could use something like this:
```ini
DATABASE_URL=mysql://vw_user:vw_userpassword@localhost/vw_database?unix_socket=/var/run/mysql/mysql.sock
```

**Reply by Henry-Hiles** [reply]
Hello, has this been done? Also, does `rocket` support listening on a unix socket?

**Reply by BlackDex** [reply]
For a long time already.

And no, rocket doesn't support this

**Comment by kinchan** [comment]
Hi, great thank you again.
I understand.
I'm waiting good news future.

**Reply by joel-bourquard** [reply]
Hi @kinchan ,
Works for me with Vaultwarden 1.34.3:
`DATABASE_URL: "mysql://username:password@localhost/vaultwarden?unix_socket=/run/mysqld/mysqld.sock"`
